src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -447,7 +153,7 @@ public static ArgumentValue CoerceValue(IGraphType type, IValue input, Variables
             {
                 if (!(input is ObjectValue objectValue))
                 {
-                    return ArgumentValue.NullLiteral;
```

Comment:
@Shane32 Why so? I believe this was a bug. See `fail_on_incorrect_value` test. I renamed it from `does_not_use_incorrect_value`. Now such query fails.

Reply:
I agree.